perceval/backends/core/gitter.py

```diff
@@ -0,0 +1,325 @@
+
+        return items
+
+    def fetch_items(self, category, **kwargs):
+        """Fetch the messages
+
+        :param category: the category of items to fetch
+        :param kwargs: backend arguments
+
+        :returns: a generator of items
+        """
+        from_date = kwargs['from_date']
+        logger.debug("Get Gitter message paginated items of room: %s from date: %s", self.room, from_date)
+        from_date = str_to_datetime(from_date).timestamp()
+        fetching = True
+        before_id = None
+        num_msgs = 0
+        self.room_id = self.client.get_room_id()
+        logger.debug("Room Id of room: %s is %s", self.room, self.room_id)
+        page = 0
+        while fetching:
+            page += 1
+            logger.debug("Page: %i" % page)
+            message_group = self.client.message_page(self.room_id, before_id)
+
+            if not message_group:
+                fetching = False
+                continue
+
+            message_group = message_group.json()
+            for raw_message in message_group:
+
+                if str_to_datetime(raw_message['sent']).timestamp() > from_date:
+                    num_msgs += 1
+                    yield raw_message
+                else:
+                    fetching = False
+
+            before_id = message_group[0]['id']
+
+        logger.debug("Fetch process completed: %s message fetched", num_msgs)
+
+    @classmethod
+    def has_archiving(cls):
+        """Returns whether it supports archiving items on the fetch process.
+
+        :returns: this backend supports items archive
+        """
+        return True
+
+    @classmethod
+    def has_resuming(cls):
+        """Returns whether it supports to resume the fetch process.
+
+        :returns: this backend does not support items resuming
+        """
+        return False
+
+    @staticmethod
+    def metadata_id(item):
+        """Extracts the identifier from a Gitter item."""
+
+        return item['id']
+
+    @staticmethod
+    def metadata_updated_on(item):
+        """Extracts and coverts the sent time of a message
+        from a Gitter item.
+
+        The timestamp is extracted from 'sent' field and
+        converted to a UNIX timestamp.
+
+        :param item: item generated by the backend
+
+        :returns: a UNIX timestamp
+        """
+        ts = str_to_datetime(item['sent'])
+
+        return ts.timestamp()
+
+    @staticmethod
+    def metadata_category(item):
+        """Extracts the category from a Gitter item.
+
+        This backend only generates one type of item which is
+        'message'.
+        """
+        return CATEGORY_MESSAGE
+
+    def _init_client(self, from_archive=False):
+        """Init client"""
+
+        room = urijoin(self.group, self.room)
+
+        return GitterClient(self.api_token, self.max_items, self.archive,
+                            from_archive, self.ssl_verify, room=room)
+
+
+class GitterClient(HttpClient):
```

Comment:
This class should inherits from [RateLimitHandler](https://github.com/chaoss/grimoirelab-perceval/blob/198c87e34e1acd7c8eee656471d77c812967b505/perceval/client.py#L223) too, since the API responses return rate limit information (see figue below). Such an info should be used to handle requests throttling/consumption.

![captura_277](https://user-images.githubusercontent.com/6515067/77246864-9d7f9880-6c2b-11ea-81e7-161d2b4e1d3c.png)

Example of the use of the RateLimitHandler are:
- https://github.com/chaoss/grimoirelab-perceval/blob/master/perceval/backends/core/gitlab.py#L464
- https://github.com/chaoss/grimoirelab-perceval/blob/master/perceval/backends/core/meetup.py#L362
- https://github.com/chaoss/grimoirelab-perceval/blob/master/perceval/backends/core/github.py#L571

In a nutshell, you have to:
- check if the ratelimit headers are the default ones (https://github.com/chaoss/grimoirelab-perceval/blob/master/perceval/client.py#L235). If not, you should redefine them (ex.: https://github.com/chaoss/grimoirelab-perceval/blob/master/perceval/backends/core/gitlab.py#L464)

- init the instance
  - https://github.com/chaoss/grimoirelab-perceval/blob/master/perceval/backends/core/meetup.py#L362

- redefine calculate_time_to_reset
  - https://github.com/chaoss/grimoirelab-perceval/blob/master/perceval/backends/core/meetup.py#L364
  - https://github.com/chaoss/grimoirelab-perceval/blob/master/perceval/backends/core/gitlab.py#L612
  - https://github.com/chaoss/grimoirelab-perceval/blob/master/perceval/backends/core/github.py#L625